Pilot/pilotTools.py

```diff
@@ -214,53 +214,97 @@ def listdir(directory):
     return contents
 
 
-def getFlavour(ceName):
+def getSubmitterInfo(ceName):
+    """
+    Check the environment variables to determine the type of batch system and CE used 
+    to submit the pilot being used and return this information in a tuple.
+    """
 
-    pilotReference = os.environ.get("DIRAC_PILOT_STAMP", "")
-    flavour = "DIRAC"
+    pilotReference = os.environ.get("DIRAC_PILOT_STAMP", "Unknown")
+    # Batch system taking care of the pilot
+    # Might be useful to extract the info to interact with it later on
+    batchSystemType = None
+    batchSystemJobID = None
+    batchSystemParameters = {
+        "BinaryPath": "Unknown",
+        "Host": "Unknown",
+        "InfoPath": "Unknown",
+        "Queue": "Unknown",
+    }
+    # Flavour of the pilot
+    # Inform whether the pilot was sent through SSH+batch system or a CE
+    flavour = "Unknown"
 
     # # Batch systems
 
-    # Take the reference from the Torque batch system
+    # Torque
     if "PBS_JOBID" in os.environ:
-        flavour = "SSHTorque"
-        pilotReference = "sshtorque://" + ceName + "/" + os.environ["PBS_JOBID"].split(".")[0]
+        batchSystemType = "Torque"
+        batchSystemJobID = os.environ["PBS_JOBID"]
+        batchSystemParameters["BinaryPath"] = os.environ.get("PBS_O_PATH", "Unknown")
+        batchSystemParameters["Queue"] = os.environ.get("PBS_O_QUEUE", "Unknown")
+
+        flavour = "SSH%s" % batchSystemType
+        pilotReference = "sshtorque://" + ceName + "/" + batchSystemJobID.split(".")[0]
 
-    # Take the reference from the OAR batch system
+    # OAR
     if "OAR_JOBID" in os.environ:
-        flavour = "SSHOAR"
-        pilotReference = "sshoar://" + ceName + "/" + os.environ["OAR_JOBID"]
+        batchSystemType = "OAR"
+        batchSystemJobID = os.environ["OAR_JOBID"]
+
+        flavour = "SSH%s" % batchSystemType
+        pilotReference = "sshoar://" + ceName + "/" + batchSystemJobID
 
     # Grid Engine
     if "JOB_ID" in os.environ and "SGE_TASK_ID" in os.environ:
-        flavour = "SSHGE"
-        pilotReference = "sshge://" + ceName + "/" + os.environ["JOB_ID"]
-    # Generic JOB_ID
-    elif "JOB_ID" in os.environ:
-        flavour = "Generic"
-        pilotReference = "generic://" + ceName + "/" + os.environ["JOB_ID"]
```

Comment:
Because I don't see where such a variable is set in the code (I might have just missed it).
If I understand correctly, this would happen if we use:
- `Local/SSH`CE + `Host` Batch system.
- `ARC` but no batch system under the hood (is it even possible?)

In the first case, `JOB_ID` does not seem to be set. There is `context.upper() + "_JOBID"` though.

Reply:
The first case (with `Host`) is rare but possible. Somehow that should be taken into consideration.

Reply:
So, `SSH` + `Host` was already taken into account, I just added the `Local` + `Host` case, which is going to replace `flavour = "Generic"`, unless you have a good reason to keep it.